### PR TITLE
Correctly display connection state

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-ocis</strong> https://drone.owncloud.com/owncloud/client/14479/7/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/14479/ocis/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/14479/ocis/guiReportUpload/serverlog.log

Screenshots:
- [logging_out.png](https://cache.owncloud.com/public/owncloud/client/14479/ocis/guiReportUpload/screenshots/logging_out.png)
- [login_after_loggin_out.png](https://cache.owncloud.com/public/owncloud/client/14479/ocis/guiReportUpload/screenshots/login_after_loggin_out.png)

</details>

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14479/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/14479/oc10/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/14479/oc10/guiReportUpload/serverlog.log

Screenshots:
- [logging_out.png](https://cache.owncloud.com/public/owncloud/client/14479/oc10/guiReportUpload/screenshots/logging_out.png)
- [login_after_loggin_out.png](https://cache.owncloud.com/public/owncloud/client/14479/oc10/guiReportUpload/screenshots/login_after_loggin_out.png)

</details>

**Comment by saw-jan** [comment]
The failure is due to the change in the text.

Before:
Screenshot from 2023-03-03 10-02-33

this PR:
Screenshot from 2023-03-03 10-02-42


Is it the intended change?

**Comment by saw-jan** [comment]
@TheOneRing I have adapted the tests according to the new text changes. If that was not the intended change then please remove my commit.

**Review by fmoc — CHANGES_REQUESTED** [review]

File: changelog/unreleased/10557 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Improve display of connection state
+
+Don't display `No connection configured` when disconnected but `Disconnected from:`.
+Introduce new state, `Connecting to`.
```
Comment (fmoc) [inline changelog/unreleased/10557:4]: We split the current "no connection configured" state into a "disconnected" and "connecting to" state to better explain the client's behavior to the user.

File: src/gui/accountsettings.cpp [hunk 2]
```diff
@@ -760,176 +760,152 @@ void AccountSettings::slotUpdateQuota(qint64 total, qint64 used)
...
-            ui->addButton->setToolTip(tr("Click this button to add a Space."));
-        } else {
-            ui->addButton->setToolTip(tr("Click this button to add a folder to synchronize."));
-        }
-    } else {
-        ui->addButton->setToolTip(tr("You need to be connected to add a folder."));
+    // in 2023 there should never be credentials encoded in the url, but we never know...
+    const auto safeUrl = account->url().adjusted(QUrl::RemoveUserInfo);
+    FolderMan *folderMan = FolderMan::instance();
+    for (auto *folder : folderMan->folders()) {
```
Comment (fmoc) [inline src/gui/accountsettings.cpp:769]:
```suggestion
    // in 2023 there should never be credentials encoded in the url, but we never know...
    const auto safeUrl = account->url().adjusted(QUrl::RemoveUserInfo);

    FolderMan *folderMan = FolderMan::instance();
    for (auto *folder : folderMan->folders()) {
```

**Review by fmoc — APPROVED** [review]
Played a bit with the code, seems to work.

**Pull request merged**